versioned/tiered/tiered-impl/src/test/java/org/projectnessie/versioned/impl/condition/TestUpdateClauseVisitor.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.projectnessie.versioned.impl.condition;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.projectnessie.versioned.store.Entity;
+
+/**
+ * Test basic functionality and show usage of the UpdateClauseVisitor as pertains to UpdateExpression and the UpdateClause
+ * hierarchy (ValueOfEntity, ExpressionFunction, ExpressionPath).
+ */
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class TestUpdateClauseVisitor {
+
+  private static final Visitor VISITOR = new Visitor();
+  private static final String ID_STR = "id";
+  private static final Entity ENTITY_STR = Entity.ofString(ID_STR);
+  private static final String PATH_STR = "commits";
+  private static final ExpressionPath PATH = ExpressionPath.builder(PATH_STR).build();
+
+  /**
+   * Sample interface or class into which UpdateClauses are converted.
+   */
+  interface UpdateCommand {
+    /**
+     * An enum encapsulating.
+     */
+    enum Operator {
+      // An operator to remove some part or all of an entity.
+      REMOVE,
+
+      // An operator to set some part or all of an entity.
+      SET
+    }
+
+    Operator getOperator();
+
+    ExpressionPath getPath();
+
+    class SetCommand implements UpdateCommand {
```

Comment:
Can you use Immutables for these two?

Reply:
Done, but needed to move class higher so it would generate Immutable class.

Reply:
This seems incorrect: https://github.com/immutables/immutables/issues/23 ?
Or is it the case that an immutable can only work if it's nested in a parent immutable?

Reply:
I have corrected this. I moved the interface `UpdateCommand` into the `TestUpdateClauseVisitor` class and nested the `RemoveCommand` and `SetCommand` classes under that. It all looks much better now. I think IntelliJ was not refreshed before I checked for the generated test classes in generated-test-sources.